rust/perspective-viewer/src/rust/components/column_settings_sidebar/attributes_tab.rs

```diff
@@ -0,0 +1,135 @@
+// ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+// ┃ ██████ ██████ ██████       █      █      █      █      █ █▄  ▀███ █       ┃
+// ┃ ▄▄▄▄▄█ █▄▄▄▄▄ ▄▄▄▄▄█  ▀▀▀▀▀█▀▀▀▀▀ █ ▀▀▀▀▀█ ████████▌▐███ ███▄  ▀█ █ ▀▀▀▀▀ ┃
+// ┃ █▀▀▀▀▀ █▀▀▀▀▀ █▀██▀▀ ▄▄▄▄▄ █ ▄▄▄▄▄█ ▄▄▄▄▄█ ████████▌▐███ █████▄   █ ▄▄▄▄▄ ┃
+// ┃ █      ██████ █  ▀█▄       █ ██████      █      ███▌▐███ ███████▄ █       ┃
+// ┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┫
+// ┃ Copyright (c) 2017, the Perspective Authors.                              ┃
+// ┃ ╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌ ┃
+// ┃ This file is part of the Perspective library, distributed under the terms ┃
+// ┃ of the [Apache License 2.0](https://www.apache.org/licenses/LICENSE-2.0). ┃
+// ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+
+use wasm_bindgen::JsValue;
+use yew::{function_component, html, Callback, Html, Properties};
+
+use crate::components::expression_editor::ExpressionEditor;
+use crate::components::viewer::ColumnLocator;
+use crate::config::ViewConfigUpdate;
+use crate::model::UpdateAndRender;
+use crate::renderer::Renderer;
+use crate::session::Session;
+use crate::utils::ApiFuture;
+use crate::{derive_model, html_template};
+
+#[derive(PartialEq, Clone, Properties)]
+pub struct AttributesTabProps {
+    pub selected_column: ColumnLocator,
+    pub on_close: Callback<()>,
+    pub session: Session,
+    pub renderer: Renderer,
+}
+derive_model!(Renderer, Session for AttributesTabProps);
+
+#[function_component]
+pub fn AttributesTab(p: &AttributesTabProps) -> Html {
+    let is_validating = yew::use_state_eq(|| false);
+    let on_save = yew::use_callback(
+        |v, p| {
+            match &p.selected_column {
+                ColumnLocator::Expr(None) => save_expr(v, p),
+                ColumnLocator::Expr(Some(alias)) => update_expr(alias, &v, p),
+                _ => {
+                    tracing::error!("Tried to save expression in bad state!")
```

Comment:
`AttributesTab` cannot currently be instantiated for non expression columns (which this error alludes to). We can represent this illegal state in the type system and validate it at compile time by not using `ColumnLocator` in props,  rather just `Option<String>`.